### ISSUE TITLE
Not sure wallis is correctly implemented

### ISSUE THREAD

**Issue by alex-s-gardner** [forrestfwilliams/wallis_benchmarking#1]
I see the local standard deviation is written as:
std = np.sqrt((squared_mean - mean) ** 2) * np.sqrt(kernel_sum / (kernel_sum - 1.0))

But I don't think that this is equivalent to:
std = sqrt((array - mean).^2)/N

**Comment by jhkennedy** [comment]
@alex-s-gardner in nasa-jpl/autoRIFT it's implemented as:
```
std = np.sqrt((squared_mean - mean ** 2) * np.sqrt(kernel_sum / (kernel_sum - 1.0))
```
https://github.com/nasa-jpl/autoRIFT/pull/59/files#diff-aef32708be8cb7db896244136a3694fcc729dc6cd5f03753863b2387c579dae1R77

@forrestfwilliams was curious about that as when you're using random data you can get negative values in the `sqrt` and switched it to this. from a performance standpoint there should be no difference in the two. 

But, I think both autorift ones don't match your second one...

**Comment by alex-s-gardner** [comment]
@jhkennedy  I believe the correct implementation will impact performance as the std needs to be computed in a local window... the way it's implemented now it's only the mean that's being computed on the moving window... not sure where this originated but it looks wrong

**Comment by jhkennedy** [comment]
@alex-s-gardner we took it from Yang's code that was already in autoRIFT.

**Comment by alex-s-gardner** [comment]
@jhkennedy looks like we inherited a blunder

**Comment by jhkennedy** [comment]
@alex-s-gardner I'm sure we've all passed a few on :rofl: . @forrestfwilliams using @alex-s-gardner julia code in #2 and #3, think you can rework the filters to match?

**Comment by alex-s-gardner** [comment]
@jhkennedy I'm flatter that you think I've only passed on a "few"

**Comment by jhkennedy** [comment]
@alex-s-gardner should we update the julia code to #3 (you closed it) or to what you posted in slack?
```julia
@time begin
    Im = imfilter(I, ones(w,w)/w^2);
    Is = I - Im;
    Istd = sqrt.(imfilter(Is.^2, ones(w,w)/w^2))
    Is = Is ./ Istd;
end
```

**Comment by alex-s-gardner** [comment]
@jhkennedy I opened and closed so many yesterday I must have closed the last one that had this... I'll create a new pull request

**Comment by forrestfwilliams** [comment]
Hi @alex-s-gardner, I've reimplemented the standard deviation filter with a correct approach. I've also added some tests to the `test_benchmark.py` that we can use to ensure this remains. Within this process, I've been taking a deep-dive into the world of standard deviation filters and I've found some interesting resources. Namely this [wikipedia page](https://en.wikipedia.org/wiki/Algorithms_for_calculating_variance#cite_note-Einarsson2005-1) and this [paper](http://cpsc.yale.edu/sites/default/files/files/tr222.pdf). They both mention that computing the standard deviation using SumSq and (Sum×Sum)/n can lead to floating point rounding errors that can sometimes lead to issues.
